media/admin/com_virtuemart/models/zasilkovna_src/VirtueMartModelZasilkovna/ShipmentMethod.php

```diff
@@ -97,132 +100,20 @@ function ($weightRule, $weightRuleAfter) {
-        } else {
-            if (!is_numeric($globalDefaultPrice)) {
-                $report->addError(ShipmentValidationReport::ERROR_CODE_INVALID_TYPE);
-            }
-        }
-
-        $globalMaxWeight = $this->getGlobalMaxWeight();
-        if (empty($globalMaxWeight) && !is_numeric($globalMaxWeight)) {
-            $report->addError(ShipmentValidationReport::ERROR_CODE_GLOBAL_MAX_WEIGHT_MISSING);
-        } else {
-            if (!is_numeric($globalMaxWeight)) {
-                $report->addError(ShipmentValidationReport::ERROR_CODE_INVALID_TYPE);
-            }
-        }
-
-        $weightsFE = ($this->getGlobalWeightRules() ?: []);
-        foreach ($weightsFE as $weightRule) {
-            $weightRulesReport = $this->validateWeightRule($weightRule, $globalMaxWeight);
-
-            if ($weightRulesReport->isValid() === false) {
-                $report->merge($weightRulesReport);
-                break;
-            }
-        }
-
-        $rules = ($this->getPricingRules() ?: []);
-        $countries = [];
-
-        foreach ($rules as $countryRule) {
-            if (array_key_exists($countryRule->country, $countries)) {
-                $report->addError(ShipmentValidationReport::ERROR_CODE_DUPLICATE_COUNTRIES); // multiple country definitions not allowed
-                break;
-            }
-
-            $countries[$countryRule->country] = $countryRule->country;
-
-            $countryWeightRules = ($this->getCountryWeightRules($countryRule->country) ?: []);
-            foreach ($countryWeightRules as $weightRule) {
-                $weightRulesReport = $this->validateWeightRule($weightRule, $globalMaxWeight);
-
-                if ($weightRulesReport->isValid() === false) {
-                    $report->merge($weightRulesReport);
-                    break;
-                }
-            }
-        }
-
-        $blockingCountries = ($this->getBlockingCountries() ?: []);
-        $allowedCountries = ($this->getAllowedCountries() ?: []);
-
-        $blockingCountries = array_diff($blockingCountries, $allowedCountries); // when user allowes and blocks same countries
-        $allowedCountries = array_diff($allowedCountries, $blockingCountries); // when user allowes and blocks same countries
-
-        if (!empty($allowedCountries)) {
-            $diff = array_diff($countries, $allowedCountries);
-            if (!empty($diff)) {
-                $report->addError(ShipmentValidationReport::ERROR_CODE_ALLOWED_COUNTRIES_ONLY);
-            }
-        }
-
-        if (!empty($blockingCountries)) {
-            $diff = array_diff($countries, $blockingCountries);
-            if (count($diff) !== count($countries)) {
-                $report->addError(ShipmentValidationReport::ERROR_CODE_NO_BLOCKED_COUNTRY);
-            }
-        }
-
-        return $report;
-    }
-
     /**
      * @return mixed
      */
     public function getAllowedCountries()
```

Comment:
návratový typ prosím více striktní. Prosím projít všechen kód, který zobrazuje GitHub a všude to opravit.